crates/miden-testing/src/kernel_tests/tx/test_account.rs

```diff
@@ -1265,6 +1266,241 @@ fn incrementing_nonce_twice_fails() -> anyhow::Result<()> {
+            push.9.10.11.12
+            assert_eqw.err=\"current value should be updated\"
+
+            # get_item_init should still return the initial value
+            push.0
+            exec.account::get_item_init
+            push.{expected_initial_value}
+            assert_eqw.err=\"initial value should remain unchanged\"
+        end
+        ",
+        expected_initial_value = &AccountStorage::mock_item_0().slot.value(),
+    );
+
+    tx_context.execute_code(&code).unwrap();
+
+    Ok(())
+}
+
+#[test]
+fn test_get_map_item_init() -> miette::Result<()> {
+    let account = AccountBuilder::new(ChaCha20Rng::from_os_rng().random())
+        .with_auth_component(Auth::IncrNonce)
+        .with_component(MockAccountComponent::with_slots(vec![AccountStorage::mock_item_2().slot]))
+        .build_existing()
+        .unwrap();
+
+    let tx_context = TransactionContextBuilder::new(account).build().unwrap();
+
+    // Use the first key-value pair from the mock storage
+    let (initial_key, initial_value) = STORAGE_LEAVES_2[0];
+    let new_key = Word::from([201, 202, 203, 204u32]);
+    let new_value = Word::from([301, 302, 303, 304u32]);
+
+    let code = format!(
+        "
+        use.$kernel::prologue
+        use.mock::account->mock_account
+
+        begin
+            exec.prologue::prepare_transaction
+
+            # get initial value from map
+            push.{initial_key}
+            push.0
+            call.mock_account::get_map_item_init
+            push.{initial_value}
+            assert_eqw.err=\"initial map value should match expected\"
+
+            # add a new key-value pair to the map
+            push.{new_value}
+            push.{new_key}
+            push.0
+            call.mock_account::set_map_item dropw dropw
+
+            # get_map_item should return the new value
+            push.{new_key}
+            push.0
+            call.mock_account::get_map_item
+            push.{new_value}
+            assert_eqw.err=\"current map value should be updated\"
+
+            # get_map_item_init should still return the initial value for the initial key
+            push.{initial_key}
+            push.0
+            call.mock_account::get_map_item_init
+            push.{initial_value}
+            assert_eqw.err=\"initial map value should remain unchanged\"
+
+            # get_map_item_init for the new key should return empty word (default)
+            push.{new_key}
+            push.0
+            call.mock_account::get_map_item_init
+            push.0.0.0.0
+            assert_eqw.err=\"new key should have empty initial value\"
+
+            dropw dropw
+        end
+        ",
+        initial_key = &initial_key,
+        initial_value = &initial_value,
+        new_key = &new_key,
+        new_value = &new_value,
+    );
+
+    tx_context.execute_code(&code).unwrap();
+
+    Ok(())
+}
+
+#[test]
+fn test_get_item_init_and_get_map_item_init_with_foreign_account() -> miette::Result<()> {
```

Comment:
Nit: If we use `anyhow::Result` we should be able to turn most unwraps into `?`.
`miette::Result` only makes sense when using `TransactionContext::execute_code` because it returns an error type that implements `Diagnostic` (one from miden-vm). When using miden-base APIs on the other hand, none of them implement `Diagnostic` so we should use `anyhow`.